test/lint/lint-circular-dependencies.sh

```diff
@@ -11,6 +11,7 @@ export LC_ALL=C
 EXPECTED_CIRCULAR_DEPENDENCIES=(
     "chainparamsbase -> util/system -> chainparamsbase"
     "index/txindex -> validation -> index/txindex"
+    "index/coinstatsindex -> node/coinstats -> index/coinstatsindex"
```

Comment:
Could move `CCoinsStats` and related to src/coinstats.h to avoid this.

Reply:
Sorry for the late reply, I am not sure if I can follow this: you mean src/node/coinstats.h (file was moved not too long ago I think), right? But the struct `CCoinsStats` is already in there.

Reply:
I agree but I don't see how to achieve this with some refactor if it's done right. It's is on the top of my TODO list to resolve some circular dependencies but I think it's out of the scope of this PR.